packages/model-viewer/src/three-components/TextureUtils.ts

```diff
@@ -173,80 +156,235 @@ export default class TextureUtils extends EventDispatcher {
+    /** tempTarget.dispose(); */
   }
 
-  async dispose() {
-    const allTargetsLoad: Array<Promise<WebGLRenderTarget>> = [];
+  private halfblur(
+      targetIn: WebGLCubeRenderTarget, targetOut: WebGLCubeRenderTarget,
+      sigmaRadians: number, direction: 'latitudinal'|'longitudinal') {
+    // Number of standard deviations at which to cut off the discrete
+    // approximation.
+    const STANDARD_DEVIATIONS = 3;
+
+    const pixels = targetIn.width;
+    const radiansPerPixel = isFinite(sigmaRadians) ?
+        Math.PI / (2 * pixels) :
+        2 * Math.PI / (2 * MAX_SAMPLES - 1);
+    const sigmaPixels = sigmaRadians / radiansPerPixel;
+    const samples = isFinite(sigmaRadians) ?
+        1 + Math.floor(STANDARD_DEVIATIONS * sigmaPixels) :
+        MAX_SAMPLES;
+
+    if (samples > MAX_SAMPLES) {
+      console.warn(`sigmaRadians, ${
+          sigmaRadians}, is too large and will clip, as it requested ${
+          samples} samples when the maximum is set to ${MAX_SAMPLES}`);
+    }
 
-    // NOTE(cdata): We would use for-of iteration on the maps here, but
-    // IE11 doesn't have the necessary iterator-returning methods. So,
-    // disposal of these render targets is kind of convoluted as a result.
+    const weights = [];
+    let sum = 0;
 
-    this.environmentMapCache.forEach((targetLoads) => {
-      allTargetsLoad.push(targetLoads);
-    });
+    for (let i = 0; i < MAX_SAMPLES; ++i) {
+      const x = i / sigmaPixels;
+      const weight = Math.exp(-x * x / 2);
+      weights.push(weight);
 
-    this.environmentMapCache.clear();
+      if (i == 0) {
+        sum += weight;
 
-    for (const targetLoads of allTargetsLoad) {
-      try {
-        const target = await targetLoads;
-        target.dispose();
-      } catch (e) {
-        // Suppress errors, so that all render targets will be disposed
+      } else if (i < samples) {
+        sum += 2 * weight;
       }
     }
 
+    for (let i = 0; i < weights.length; i++) {
+      weights[i] = weights[i] / sum;
+    }
+
+    const blurUniforms = this.blurMaterial!.uniforms;
+    blurUniforms['envMap'].value = targetIn.texture;
+    blurUniforms['samples'].value = samples;
+    blurUniforms['weights'].value = weights;
+    blurUniforms['latitudinal'].value = direction === 'latitudinal';
+    blurUniforms['dTheta'].value = radiansPerPixel;
+
+    const cubeCamera = new CubeCamera(0.1, 100, targetOut);
+    cubeCamera.update(this.threeRenderer, this.blurScene!);
+  }
+
+  private getBlurShader(maxSamples: number) {
+    const weights = new Float32Array(maxSamples);
+    const poleAxis = new Vector3(0, 1, 0);
+    const shaderMaterial = new ShaderMaterial({
+
+      name: 'SphericalGaussianBlur',
+
+      defines: {'n': maxSamples},
+
+      uniforms: {
+        'envMap': {value: null},
+        'samples': {value: 1},
+        'weights': {value: weights},
+        'latitudinal': {value: false},
+        'dTheta': {value: 0},
+        'poleAxis': {value: poleAxis}
```

Comment:
This shader is a simplification of the one I built for PMREM, meant  to work on regular cube maps rather than cube UV.

Reply:
As already discussed. Yes yes yes! 🤓